6_nodejs/week2/index.js

```diff
@@ -0,0 +1,94 @@
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+const { readFileSync } = require("fs");
+const { parse } = require('path');
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/", (req, res) => {
+    res.send("This is a search engine");
+});
+
+app.get("/search", (req, res) => {
+    const query = req.query.q
+    const data = readFileSync('./documents.json');
+    const parsedData = JSON.parse(data);
+
+    if (!query) {
+        res.send(parsedData)
+    }
+
+    for (const document of parsedData) {
```

Comment:
Nice that you used a creative way to solve the problem. You could also consider using **.some** in the array, this way is much faster.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some